### PR TITLE
Improvements and changes in reported cases

### PR DESCRIPTION
**Pull request by MohsenSaeidiDev**

### CODE REVIEW

File: src/assets/bidashboard/dist/css/custom.css [hunk 1]
```diff
@@ -93,4 +93,21 @@
...
+    white-space: nowrap;
+    text-overflow: ellipsis;
+}
+.link-button{
+    border: 0;
+    background: none;
+    cursor: pointer;
+    outline: none!important;
+}
+.btn-red{
```
Comment (Sadi01) [inline src/assets/bidashboard/dist/css/custom.css]: use bootstrap classes, class="alert alert-danger"

File: src/controllers/ReportWidgetController.php [hunk 2]
```diff
@@ -330,4 +312,52 @@ public function actionModalShowChart($id, $field, $start_range = null, $end_rang
...
+            ->all();
+
+        $pages = ReportPage::find()
+            ->select(['id', 'title'])
+            ->where(['id' => ReportPageWidget::find()
+                ->select('page_id')
+                ->where(['widget_id' => $model->id])
+                ->distinct()
+                ->column()])
+            ->all();
```
Comment (Sadi01) [inline src/controllers/ReportWidgetController.php]: use model relations

File: src/migrations/m240302_082109_add_slave_id_column_to_report_year_table.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+use yii\db\Migration;
+
+/**
+ * Handles adding columns to table `{{%report_year}}`.
+ */
+class m240302_082109_add_slave_id_column_to_report_year_table extends Migration
```
Comment (Sadi01) [inline src/migrations/m240302_082109_add_slave_id_column_to_report_year_table.php:8]: Consider slave_id in migration, use report_widget table, slave_id column

File: src/models/ReportBoxWidgets.php [hunk 4]
```diff
@@ -158,6 +160,8 @@ public static function getFormattedValue($format, $value)
         return match ($format) {
             self::FORMAT_NUMBER => Yii::$app->formatter->asInteger($value),
             self::FORMAT_CURRENCY => Yii::$app->formatter->asCurrency($value),
+            self::FORMAT_GRAM => number_format($value) . ' ' . Yii::t('biDashboard', 'Gram'),
+            self::FORMAT_KILOGRAM =>Yii::$app->formatter->asWeight($value),
```
Comment (Sadi01) [inline src/models/ReportBoxWidgets.php]: $value / 1000 . KiloGram

File: src/models/ReportPage.php [hunk 5]
```diff
@@ -143,6 +145,8 @@ public static function itemAlias($type, $code = NULL)
             'Format' => [
                 self::FORMAT_CURRENCY => Yii::t('biDashboard', 'Currency'),
                 self::FORMAT_NUMBER => Yii::t('biDashboard', 'Number'),
+                self::FORMAT_GRAM => Yii::t('biDashboard', 'Gram'),
+                self::FORMAT_KILOGRAM => Yii::t('biDashboard', 'Kilo Gram'),
```
Comment (Sadi01) [inline src/models/ReportPage.php:149]: $value / 1000 . KiloGram

File: src/models/ReportPageWidget.php [hunk 6]
```diff
@@ -92,6 +95,10 @@ public function getFormattedValue($value)
                 return Yii::$app->formatter->asInteger($value);
             case self::FORMAT_CURRENCY:
                 return Yii::$app->formatter->asCurrency($value);
+            case self::FORMAT_GRAM:
+                return number_format($value) . ' ' . Yii::t('biDashboard', 'Gram');
+            case self::FORMAT_KILOGRAM:
+                return Yii::$app->formatter->asWeight($value);
```
Comment (Sadi01) [inline src/models/ReportPageWidget.php]: $value / 1000 . KiloGram

File: src/models/ReportPageWidget.php [hunk 7]
```diff
@@ -92,6 +95,10 @@ public function getFormattedValue($value)
                 return Yii::$app->formatter->asInteger($value);
```
Comment (Sadi01) [inline src/models/ReportPageWidget.php:96]: replace with utility function in helper class

File: src/views/report-dashboard/box_table.php [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    <div class="card-header d-flex align-items-center justify-content-between px-3">
+        <span><?= $box->title ?? 'عنوان باکس' ?> | <span class="font-12 font-normal">(<?= ReportBox::itemAlias('RangeType', $box->range_type) ?>)</span></span>
+        <div class="d-flex align-items-center">
+            <div class="d-flex align-items-center">
+                <div class="d-flex mr-1">
+                    <?php if ($box->range_type == ReportBox::RANGE_TYPE_DAILY): ?>
+                        <div>
+                            <select name="month" class="form-control rounded-md" id="select_month_<?= $box->id ?>">
+                                <?php for ($i = 1; $i <= 12; $i++): ?>
+                                    <option value="<?= $i ?>" <?= $box->lastRunDate['month'] == $i ? 'selected' : '' ?> ><?= $pdate->jdate_words(['mm' => $i])['mm'] ?></option>
```
Comment (Sadi01) [inline src/views/report-dashboard/box_table.php]: Yii::$app->formatter->asDatetime($i, 'mm');

File: src/views/report-dashboard/view.php [hunk 9]
```diff
@@ -28,11 +26,6 @@
 $this->title = $model->title;
 $pdate = Yii::$app->pdate;
 
-HighchartsAsset::register($this)->withScripts(['modules/stock', 'modules/exporting', 'modules/drilldown']);
```
Comment (Sadi01) [inline src/views/report-dashboard/view.php:31]: check remote or local?

**Review by Sadi01 — CHANGES_REQUESTED** [review]

File: src/assets/bidashboard/dist/css/custom.css (hunk 1 above)
Comment (MohsenSaeidiDev) [reply src/assets/bidashboard/dist/css/custom.css]: used!

File: src/migrations/m240302_082109_add_slave_id_column_to_report_year_table.php (hunk 3 above)
Comment (MohsenSaeidiDev) [reply src/migrations/m240302_082109_add_slave_id_column_to_report_year_table.php:8]: Done!

File: src/models/ReportBoxWidgets.php (hunk 4 above)
Comment (MohsenSaeidiDev) [reply src/models/ReportBoxWidgets.php]: Done!

File: src/models/ReportPage.php (hunk 5 above)
Comment (MohsenSaeidiDev) [reply src/models/ReportPage.php:149]: Done!

File: src/models/ReportPageWidget.php (hunk 6 above)
Comment (MohsenSaeidiDev) [reply src/models/ReportPageWidget.php]: Done!

File: src/models/ReportPageWidget.php (hunk 7 above)
Comment (MohsenSaeidiDev) [reply src/models/ReportPageWidget.php:96]: in the future!

File: src/views/report-dashboard/box_table.php (hunk 8 above)
Comment (MohsenSaeidiDev) [reply src/views/report-dashboard/box_table.php]: in the future!

Comment (Sadi01) [reply src/views/report-dashboard/box_table.php]: Yii::$app->formatter->asDatetime($i, 'MMMM')

File: src/models/ReportBox.php [hunk 10]
```diff
@@ -93,25 +100,40 @@ public function rules()
...
 
     public function scenarios()
     {
         return [
-            self::SCENARIO_DEFAULT => ['dashboard_id', 'display_type', 'range_type', 'title', 'description', 'slave_id', 'chart_type'],
-            self::SCENARIO_CREATE => ['dashboard_id', 'display_type', 'range_type', 'title', 'description', 'slave_id', 'chart_type'],
-            self::SCENARIO_UPDATE => ['dashboard_id', 'display_type', 'title', 'description', 'slave_id', 'chart_type'],
+            self::SCENARIO_DEFAULT => ['order', 'dashboard_id', 'display_type', 'range_type', 'title', 'description', 'slave_id', 'chart_type'],
+            self::SCENARIO_CREATE => ['order', 'dashboard_id', 'display_type', 'range_type', 'title', 'description', 'slave_id', 'chart_type'],
+            self::SCENARIO_UPDATE => ['order', 'dashboard_id', 'display_type', 'title', 'description', 'slave_id', 'chart_type'],
```
Comment (Sadi01) [inline src/models/ReportBox.php:114]: display_order

**Review by Sadi01 — APPROVED** [review]

**Pull request merged**